### ISSUE TITLE
Incorrect volume / chapter numbering when using raw images

### ISSUE THREAD

**Issue by JonSchram** [Kareadita/Kavita#2762]
### What happened?

I only just installed Kavita for the first time, so I'm not sure if this is just an issue with my naming, but manga volume and chapter numbers aren't working quite right with raw images.

I have a folder structure like this (everything uses raw images named "Page #"):
```
root/
     - One/
           - One Punch Man/
                    - Volume 1/
                          - Chapter 1 (One Punch)/
                                   - Page 1.jpg
                                   - Page 2.jpg
                                    ...
                           - Chapter 2 (Crab and Job Hunting)/
                                   - Page 1.jpg
                                    ...
                               ...
                           - Chapter 08/
                           - Chapter 08.5 (200 Yen Volume 1 Extras)/
                     - Volume 2/
                        ...
```

I have been following the guidance for [raw images](https://wiki.kavitareader.com/en/guides/managing-your-files/raw-images) and [manga](https://wiki.kavitareader.com/en/guides/managing-your-files/manga) and have everything in a folder for the author, then manga name.

In Kavita, if I view the manga details and go to the info tab, I only see volume 0 with all pages in it. Looking around the documentation, it seems that there should be a Volumes tab for the manga but I don't see anything here either.

If I view details for chapter 8, I see files for chapter 8 and chapter 8.5 in the info tab. If I try to read chapter 8, nothing in chapter 8.5 will render (it will only show the last page in chapter 8). It also shows a popup that says "400 error / There was an issue saving progress."

### What did you expect?

I expected chapters to be grouped into volumes in the . I also expected bonus chapters with fractional numbers like 8.5 to appear as a separate chapter in the volume. I also expected to be able to view pages in these volumes.

I found bug #2277 about fractional chapters so I thought this was a supported use case, but that uses .cbz files so maybe the bug wasn't fixed for raw images.

### Kavita Version Number - Don't see your version number listed? Then your install is out of date. Please update and see if your issue still persists.

0.7.14 - Stable

### What operating system is Kavita being hosted from?

Docker (Other)

### If the issue is being seen on Desktop, what OS are you running where you see the issue?

Windows

### If the issue is being seen in the UI, what browsers are you seeing the problem on?

Chrome

### If the issue is being seen on Mobile, what OS are you running where you see the issue?

None

### Relevant log output

```shell
[Kavita] [2024-03-04 11:20:19.729 -05:00  316] [Information] Serilog.AspNetCore.RequestLoggingMiddleware HTTP GET "/api/reader/image?chapterId=2549&apiKey=******REDACTED******&page=33" responded 200 in 188.1633 ms
[Kavita] [2024-03-04 11:20:19.729 -05:00  359] [Information] API.Controllers.ReaderController Fetching Page 34 on Chapter 2549
[Kavita] [2024-03-04 11:20:19.729 -05:00  304] [Error] Microsoft.EntityFrameworkCore.Update An exception occurred in the database while saving changes for context type 'API.Data.DataContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 5: 'database is locked'.
   at Microsoft.Data.Sqlite.SqliteDataRecord.Dispose()
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at System.Data.Common.DbDataReader.NextResultAsync(CancellationToken cancellationToken)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ConsumeAsync(RelationalDataReader reader, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ConsumeAsync(RelationalDataReader reader, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 5: 'database is locked'.
   at Microsoft.Data.Sqlite.SqliteDataRecord.Dispose()
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at System.Data.Common.DbDataReader.NextResultAsync(CancellationToken cancellationToken)
--- End of stack trace from previous location ---
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ConsumeAsync(RelationalDataReader reader, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ConsumeAsync(RelationalDataReader reader, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
```

### Additional Notes

I am running Kavita server in Docker on Synology.

**Comment by DieselTech** [comment]
> I found bug https://github.com/Kareadita/Kavita/issues/2277 about fractional chapters so I thought this was a supported use case, but that uses .cbz files so maybe the bug wasn't fixed for raw images.

Kavita has different readers and sets of rules depending on the type of library used. It just wasn't reported before with raw images because most people use .cbz's to store their files. The raw image library type isn't used by many people so it likely just went unreported. 

As for the database locked errors that aren't updating progress, you can do what the FAQ entry here describes to fix that:
https://wiki.kavitareader.com/en/faq#q-im-seeing-database-is-locked-errors-in-my-logs

**Comment by majora2007** [comment]
Just an update, I fixed this and it will be released in v0.8.0 as long as you use Image Library type. I didn't receive feedback when I originally implemented the Image library type (as users requested I keep loose leaf image support), so I assumed it was working. I can see now that it wasn't working appropriately.
